src/Azure/Provider.php

```diff
@@ -91,17 +98,45 @@ protected function mapUserToObject(array $user)
      */
     protected function getTokenFields($code)
     {
-        return array_merge(parent::getTokenFields($code), [
+        $fields = [
             'grant_type' => 'authorization_code',
-            'resource'   => 'https://graph.windows.net',
-        ]);
+            'client_id' => $this->clientId,
+            'client_secret' => $this->clientSecret,
+            'code' => $code,
+            'redirect_uri' => $this->redirectUrl,
+        ];
+
+        if ($this->usesPKCE()) {
+            $fields['code_verifier'] = $this->request->session()->pull('code_verifier');
+        }
+
+        return $fields;
     }
 
     /**
      * {@inheritdoc}
      */
     public static function additionalConfigKeys()
     {
-        return ['tenant'];
+        return ['tenant', 'logout_url'];
     }
+
+    /**
+     * Get the access token response for the given code.
+     *
+     * @param  string  $code
+     * @return array
+     */
+    public function getAccessTokenResponse($code)
+    {
+
+        $response = $this->getHttpClient()->post($this->getTokenUrl(), [
+            'headers' => ['Accept' => 'application/json'],
```

Comment:
Can you replace headers, form_params, and proxy by RequestOptions constants instead

Reply:
done